### PR TITLE
[client] add XDG Base Directory specification compliance for config file loading

### PR DESCRIPTION
**Pull request by thomasqueirozb**
…ile loading

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -85,9 +85,10 @@ The syntax is simple: `module:name=value`, for example:
 
 ### Setting options via configuration files
 
-By default the application will look for and load the config files in the following locations
+By default the application will load the config file in /etc/looking-glass-client.ini and then look for one user config file in the following locations
```
Comment (gnif) [inline client/README.md:88]: There is no need to change the wording here, please retain the list.

File: client/src/config.c [hunk 2]
```diff
@@ -433,31 +435,99 @@ void config_init(void)
...
+    }
+    free(localFile);
+  }
+
+  if (!user_config_loaded)
   {
-    DEBUG_INFO("Loading config from: %s", localFile);
-    if (!option_load(localFile))
+    char * localFile;
+    alloc_sprintf(&localFile, "%s/.looking-glass-client.ini", pw->pw_dir);
```
Comment (gnif) [inline client/src/config.c:516]: Missing free on success

**Review by gnif — CHANGES_REQUESTED** [review]
Please see the requested changes. Also fix the commit message to a maximum width of 72 characters, ie:

`[client] config: load configuration as per XDG spec`

Also please provide a summary of the changes and the rationale behind them.

File: client/src/config.c [hunk 3]
```diff
@@ -433,31 +435,99 @@ void config_init(void)
...
+}
+
 bool config_load(int argc, char * argv[])
 {
+  bool err = false;
   // load any global options first
-  struct stat st;
-  if (stat("/etc/looking-glass-client.ini", &st) >= 0)
+
+  if (!load_config_from_file("/etc/looking-glass-client.ini", &err) || err)
```
[the remark this answers is not in the document]
Comment (thomasqueirozb) [reply client/src/config.c:458]: I was just trying to replicate the previous behavior. config_load would return false if /etc/looking-glass-client.ini or ~/.looking-glass-client.ini failed to load.

https://github.com/gnif/LookingGlass/blob/30c3b399f28bdd2afb2d574a997a1dd0924fe892/client/src/config.c#L440-L445
https://github.com/gnif/LookingGlass/blob/30c3b399f28bdd2afb2d574a997a1dd0924fe892/client/src/config.c#L450-L459

**Comment by gnif** [comment]
Closing this outdated PR.... note though that implementing the XDG standard is on the list for the next release.

**Pull request closed without merging**